desc/magnetic_fields.py

```diff
@@ -383,15 +385,44 @@ class ScaledMagneticField(_MagneticField):
 
     _io_attrs = _MagneticField._io_attrs_ + ["_field", "_scalar"]
 
-    def __init__(self, scalar, field):
-        assert np.isscalar(scalar), "scalar must actually be a scalar value"
+    def __init__(self, scale, field):
+        assert np.isscalar(scale) or len(scale) == 1, "scale must be a scalar value"
         assert isinstance(
             field, _MagneticField
         ), "field should be a subclass of MagneticField, got type {}".format(
             type(field)
         )
-        self._scalar = scalar
-        self._field = field
+        object.__setattr__(self, "_scale", scale)
```

Comment:
why this syntax and not the usual `self._scale = scale` with a default value for `scale` in the init?

Reply:
It's needed because I'm overriding the `getattr` and `setattr` behavior so that the scaled field "acts like" the underlying field for most purposes (ie the optimizable attributes are seen as a pass through)